### PR TITLE
Added preliminary FileInformationFactory implementation

### PR DESCRIPTION
**Pull request by KallanBao**
## Description
* Added preliminary FileInformationFactory implementation. Factory can now classify different file types and build CodeFileInformation, ConfigFileInformation, StaticFileInformation, ViewFileInformation, and ProjectFileInformation objects based on the file extension

## Supplemental testing
Basic unit testing was added for both the Build and BuildMany functions, but more extensive testing to handle edge cases/special files may be needed

---
*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: src/CTA.WebForms2Blazor/Factories/FileInformationFactory.cs [hunk 1]
```diff
@@ -6,9 +6,9 @@
 
 namespace CTA.WebForms2Blazor.Factories
 {
-    public static class FileInformationFactory
+    public class FileInformationFactory
     {
-        public static FileInformation Build(FileInfo document)
+        public FileInformation Build(FileInfo document, string sourceProjectPath)
```
Comment (jonlouie) [inline src/CTA.WebForms2Blazor/Factories/FileInformationFactory.cs:11]: `Build()` method can likely be made static since it does not rely on any variables within the `FileInformationFactory` class itself.

File: src/CTA.WebForms2Blazor/Factories/FileInformationFactory.cs [hunk 2]
```diff
@@ -18,12 +18,34 @@ public static FileInformation Build(FileInfo document)
...
+            {
+                fi = new StaticFileInformation(relativePath);
+            }
+
+
+            return fi;
         }
 
-        public static IEnumerable<FileInformation> BuildMany(IEnumerable<FileInfo> documents)
+        public IEnumerable<FileInformation> BuildMany(IEnumerable<FileInfo> documents, string sourceProjectPath)
```
Comment (jonlouie) [inline src/CTA.WebForms2Blazor/Factories/FileInformationFactory.cs:46]: This method can be made static for same reason as above.

File: src/CTA.WebForms2Blazor/MigrationManager.cs [hunk 3]
```diff
@@ -18,18 +18,20 @@ public class MigrationManager
         private WebFormsProjectAnalyzer _projectAnalayzer;
         private BlazorProjectBuilder _projectBuilder;
         private IEnumerable<FileInformation> _fileInformationCollection;
+        private FileInformationFactory fileFactory;
```
Comment (jonlouie) [inline src/CTA.WebForms2Blazor/MigrationManager.cs:21]: If factory build methods are made static, we won't need to hold this variable here

File: tst/CTA.WebForms2Blazor.Tests/FileInformationFactoryTests.cs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    class FileInformationFactoryTests
+    {
+        private const string TEST_FILES_DIRECTORY_PATH = "TestingArea/TestFiles";
+
+        private string _testProjectPath;
+        private string testCodeFilePath;
+        private string testConfigFilePath;
+        private string testStaticFilePath;
+        private string testViewFilePath;
+        private string testProjectFilePath;
```
Comment (jonlouie) [inline tst/CTA.WebForms2Blazor.Tests/FileInformationFactoryTests.cs:20]: nit 1: Rename constant to use pascal case (TestFilesDirectoryPath) to remain consistent with our team's adopted style

nit 2: rename private class variables to "_variableName" for consistency

File: tst/CTA.WebForms2Blazor.Tests/FileInformationFactoryTests.cs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+        private Factories.FileInformationFactory _fileFactory;
+
+        [OneTimeSetUp]
+        public void OneTimeSetup()
+        {
+            var workingDirectory = Environment.CurrentDirectory;
+            _testProjectPath = Directory.GetParent(workingDirectory).Parent.Parent.FullName;
+            string _testFilesPath = Path.Combine(_testProjectPath, TEST_FILES_DIRECTORY_PATH);
```
Comment (jonlouie) [inline tst/CTA.WebForms2Blazor.Tests/FileInformationFactoryTests.cs:29]: nit: for consistency, replace `string _testFilesPath` with `var testFilesPath`

File: tst/CTA.WebForms2Blazor.Tests/FileInformationFactoryTests.cs [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+            FileInformationModel.FileInformation codeFileObj = _fileFactory.Build(new FileInfo(testCodeFilePath), _testProjectPath);
+            FileInformationModel.FileInformation configFileObj = _fileFactory.Build(new FileInfo(testConfigFilePath), _testProjectPath);
+            FileInformationModel.FileInformation staticFileObj = _fileFactory.Build(new FileInfo(testStaticFilePath), _testProjectPath);
+            FileInformationModel.FileInformation viewFileObj = _fileFactory.Build(new FileInfo(testViewFilePath), _testProjectPath);
+            FileInformationModel.FileInformation projectFileObj = _fileFactory.Build(new FileInfo(testProjectFilePath), _testProjectPath);
+
+            //Console.WriteLine(codeFileObj.GetType());
+            //Console.WriteLine(configFileObj.GetType());
+            //Console.WriteLine(staticFileObj.GetType());
+            //Console.WriteLine(viewFileObj.GetType());
```
Comment (jonlouie) [inline tst/CTA.WebForms2Blazor.Tests/FileInformationFactoryTests.cs:55]: Remove unused code

File: tst/CTA.WebForms2Blazor.Tests/FileInformationFactoryTests.cs [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+            testStaticFilePath = Path.Combine(_testFilesPath, "SampleStaticFile.png");
+            testViewFilePath = Path.Combine(_testFilesPath, "SampleViewFile.aspx");
+            testProjectFilePath = Path.Combine(_testFilesPath, "SampleProjectFile.csproj");
+        }
+
+        [SetUp]
+        public void Setup()
+        {
+            _fileFactory = new Factories.FileInformationFactory();
+        }
```
Comment (jonlouie) [inline tst/CTA.WebForms2Blazor.Tests/FileInformationFactoryTests.cs:41]: If factory methods are made static, can remove this setup function altogether

**Review by jonlouie — CHANGES_REQUESTED** [review]
Mostly minor comments, otherwise looks good for the most part

File: src/CTA.WebForms2Blazor/Factories/FileInformationFactory.cs (hunk 1 above)
Comment (KallanBao) [reply src/CTA.WebForms2Blazor/Factories/FileInformationFactory.cs:11]: Regarding the decision to make the FileInformationFactory class non static vs static, Emery and I had a conversation about how it might be useful in the future to initialize different information in the FileInformationFactory instances and thus non static but I'll go over it again and see if this is really necessary

Comment (jonlouie) [reply src/CTA.WebForms2Blazor/Factories/FileInformationFactory.cs:11]: Sounds good. No need to revisit if you guys have already discussed this point.

Comment (comsemer) [reply src/CTA.WebForms2Blazor/Factories/FileInformationFactory.cs:11]: To add to this, I think sourceProjectPath will be the same whenever the Build (and BuildMany) function is used so I think it might be best to move this into the constructor of the factory, assign it to a private variable, then use that when calculating the relative path to create FileInformation objects

**Review by comsemer — APPROVED** [review]
Looks good to me

**Review by jonlouie — APPROVED** [review]

**Review by Eruanion — COMMENTED** [review]
Looks good thx.
